### PR TITLE
Reddit Tweets Prediction

### PR DESCRIPTION
**Pull request by aryantiwari10**
## Related Issue 

- Info about the related issue 

- [x] LGM-SOC'21 Participant 
- [ ] DevIncept Participant
- [x] Contributor

Closes: #539

### Describe the add-ons or changes you've made

PROJECT TITLE : Reddit Tweets Prediction

GOAL : To predict about the different reddit tweets in detail.

DATASET : https://www.kaggle.com/youben/tweets-sentiment-analysis

WHAT I HAD DONE : In this project first I first performed a exploratory data analysis on the given dataset which includes of data cleaning , data manipulation, data preprocessing , data visualization and after that we used NLP for the stopwords removal and cleaning, after using NLP we did the model building using different machine learning classification and regression algorithms and then predicted the accuracy of every model . In the model prediction part we used different machine learning algorithms . In each algorithm I had included the accuracy score , training score , confusion matrix, classification report . While in the EDA part I have included different plots for the different visualizations of our dataset .Some plots which we used for visualizing the dataset are Histogram , Barplot , Boxplot, Heatmap , Scatter plot , Pairplot , Jointplot etc.

ACCURACIES OF DIFFERENT MODELS ARE:

Logistic Regression= 83.70 %

Decision Tree Classifier= 58 %

Naiye Bayes= 88 %

Ada Boost Classifier = 68 %

LIBRARIES:

PANDAS

NUMPY

MATPLOTLIB

SEABORN

SCIPY

SKLEARN

Keras

Tensorflow

Nltk

CONCLUSION :

From executing all the algorithms , Naive Bayes got a training accuracy of 88% , then logistic regression with 84% training accuracy score which is quite well for the given dataset

The accuracy of other models can be increased further by HyperTuning.

## Type of change

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Describe how it has been tested
Describe how have you verified the changes made

## Checklist:

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

## Screenshots (Only for Front End and UI/UX Designers)

 Original           | Updated
 :--------------------: |:--------------------:
 Original Screenshot | Updated Screenshot |

### CODE REVIEW

**Review by prathimacode-hub — APPROVED** [review]
Your PR is approved. Yayy. Awesome work and clean PR. 🎇📌@aryantiwari10
